Essentials/src/com/earth2me/essentials/signs/EssentialsSign.java

```diff
@@ -435,6 +437,33 @@ protected final Trade getTrade(final ISign sign, final int index, final int decr
             return new Trade(money, ess);
         }
     }
+      
+    /**
+     * Updates the sign cost to the value of the item specified in worth.yml multiplied by a multiplier factor
+     * @param sign
+     * @param player
+     * @param ess
+     * @param multiplier multiply the worth price by this amount
+     * 
+     * @returns true if the price was changed, false otherwise
+     */
+    protected boolean updateFromWorth(final ISign sign, final User player, final IEssentials ess, final BigDecimal multiplier) throws SignException {
+        final ItemStack stack = getItemStack(getSignText(sign, 2), getIntegerPositive(sign.getLine(1)), ess);
+        final int amount = stack.getAmount();
+        BigDecimal price = ess.getWorth().getPrice(ess, stack);
+        if (price != null && amount > 0) {
+            price = price.multiply(multiplier).multiply(new BigDecimal(amount)).setScale(2, RoundingMode.UP);
+            final BigDecimal oldPrice = getMoney(getSignText(sign, 3), ess);
+            if (oldPrice == null || price.compareTo(oldPrice) != 0) {
+                final String priceString = NumberUtil.shortCurrency(price, ess);
+                sign.setLine(3, tl("signFormatWorth") + priceString);
+                sign.updateSign();
+                player.sendMessage(tl("priceChanged", amount, stack.getType().toString().toLowerCase(Locale.ENGLISH), (oldPrice == null ? tl("none") : NumberUtil.shortCurrency(oldPrice, ess)), priceString));
+                return true;
+            }
+        }
+        return false;
+  }
```

Comment:
This behaviour is specific to buy and sell signs - I'd be inclined to put this in an `EssentialsShopSign` subclass.

Reply:
Done